packages/admin/admin/src/common/buttons/save/SaveButton.tsx

```diff
@@ -1,14 +1,74 @@
 import { Check, Error, Error as ErrorIcon, Save, ThreeDotSaving } from "@comet/admin-icons";
-import { Button, ButtonClassKey, ButtonProps, ComponentsOverrides, Theme } from "@mui/material";
-import { WithStyles, withStyles } from "@mui/styles";
-import { ClassKeyOfStyles } from "@mui/styles/withStyles";
-import { ClassNameMap } from "@mui/styles/withStyles/withStyles";
+import { Button, ButtonClassKey, buttonGroupClasses, ButtonProps, ComponentsOverrides } from "@mui/material";
+import { css, styled, Theme, useThemeProps } from "@mui/material/styles";
 import * as React from "react";
 import { FormattedMessage } from "react-intl";
 
 import { messages } from "../../../messages";
 import { useSplitButtonContext } from "../split/useSplitButtonContext";
-import { SaveButtonClassKey, styles } from "./SaveButton.styles";
+
+export type SaveButtonClassKey = "root" | "saving" | "error" | "success" | "conflict" | ButtonClassKey;
+
+type OwnerState = Pick<SaveButtonProps, "variant" | "color"> & { displayState?: SaveButtonDisplayState };
+
+const Root = styled(Button, {
+    name: "CometAdminSaveButton",
+    slot: "root",
+    overridesResolver({ ownerState }: { ownerState: OwnerState }, styles) {
+        return [
+            styles.root,
+            ownerState.displayState === "saving" && styles.saving,
+            ownerState.displayState === "error" && styles.error,
+            ownerState.displayState === "success" && styles.success,
+            ownerState.displayState === "conflict" && styles.conflict,
+        ];
+    },
+})<{ ownerState: OwnerState }>(
+    ({ ownerState, theme }) => css`
+        ${ownerState.displayState === "saving" &&
+        css`
+            &:disabled {
+                ${ownerState.variant === "contained" &&
+                ownerState.color === "primary" &&
+                css`
+                    color: ${theme.palette.primary.contrastText};
+                    background-color: ${theme.palette.primary.main};
+                `};
+                ${ownerState.variant === "contained" &&
+                ownerState.color === "secondary" &&
+                css`
+                    color: ${theme.palette.secondary.contrastText};
+                    background-color: ${theme.palette.secondary.main};
+                `}
+            }
+        `}
+        ${ownerState.displayState === "error" &&
+        css`
+            &:disabled {
+                color: ${theme.palette.error.contrastText};
+                background-color: ${theme.palette.error.light};
+            }
+        `}
+        ${ownerState.displayState === "success" &&
+        css`
+            &:disabled {
+                color: ${theme.palette.success.contrastText};
+                background-color: ${theme.palette.success.light};
+            }
+        `}
```

Comment:
@jamesricky is it correct that these styles should only be applied if the button is disabled?

Reply:
Yes, at least that's how it was before:
<img width="404" alt="image" src="https://github.com/vivid-planet/comet/assets/6264317/a45d8830-b505-44eb-82a7-af6ac5656118">